### PR TITLE
# [FEATURE] Enhance and Refactor Interview Management Features

### PR DESCRIPTION
**Pull request by olhasashchuk**
# [FEATURE] Enhance and Refactor Interview Management Features

---

## Description

This pull request introduces significant improvements to the interview management system, targeting both frontend and backend functionalities. These changes address current issues with validation, responsiveness, and consistency, while enhancing user experience and streamlining code for better maintainability.

- **Problem Solved:** Enhanced interview management with intuitive features and reliable API interactions.
- **Impact:** Improved user experience, consistent data handling, and cleaner, modular code.

---

#### **Components**
- **Interview Tab:**  
  - Refactored `Interview.js` to support listing, addition, editing, and deletion of interviews.
  - Introduced `InterviewCard` for modular and reusable rendering of interview details.
  - Added dynamic support for "Online" and "In-person" interviews with Google Meet links and location fields.

- **Modals and Forms:**  
  - Updated `InterviewDeleteModal` and `InterviewForm` for better context integration and enhanced validation.  
  - Improved `InterviewForm` to handle "add" and "edit" modes seamlessly.  

- **Reusable Utilities:**  
  - Extended `MenuButtonInterview` with additional props for custom interactions.  
  - Enhanced `formatDate` and `formatRelativeTime` utilities for consistent formatting.  
  - Modified `InputField` to include `ref` support for better compatibility with forms.  

#### **UI Enhancements**
- **Responsiveness:**  
  - Enhanced layout responsiveness for mobile and desktop using MUI components.  

- **Fallback Handling:**  
  - Improved `ContactCard` with appropriate fallback for missing role, phone, or email values.  
  - Added an `EmptyState` for scenarios with no interviews.  

- **Consistency:**  
  - Adjusted padding in `JobInfo.js`, `Notes.js`, and `ManagePanel` for uniform spacing.  

### Backend

- **Validation:**  
  - Refactored validation logic in `interviewController.js` to require `type` instead of `is_virtual`.  
  - Enhanced `patchUserInterview` to gracefully handle undefined or null locations.  

- **Response Standardization:**  
  - Updated `patchUserInterview` to use `buildInterviewDto`, ensuring consistent DTO formatting.  

### API Integrations
- Added new functions in `api.js` for `GET`, `POST`, `PATCH`, and `DELETE` interview-related requests.  
- Integrated `deleteInterviewByApplicationId` and `patchInterviewByApplicationId` to streamline backend interactions.  

---

## Screenshots
Video:
https://github.com/user-attachments/assets/9c0ddf5e-b5b4-4ef7-9049-290e97bbc08a

Screenshot_3
Screenshot_2
Screenshot_1

---

## Testing Steps

1. **Frontend:**
   - Navigate to the **Applications** tab and select the **Interview** section.
   - Test the following:
     - Add an interview with valid and invalid inputs.
     - Edit an existing interview and verify the updates.
     - Delete an interview and check the updated UI.
     - Toggle between "Online" and "In-person" interviews to validate input handling.
   - Ensure proper layout and responsiveness on mobile and desktop devices.

2. **Backend:**
   - Test API endpoints for `GET`, `POST`, `PATCH`, and `DELETE` operations to verify validation and response formatting.

---

## Issues Addressed

Closes (link to Trello ticket or issue tracker).

---

## Additional Notes

- **Known Limitations:** None identified.  
- **Future Work:** Additional test coverage for edge cases in validation.

### CODE REVIEW

**Comment by olhasashchuk** [comment]
I fixed it
Screenshot_4

**Review by pankaj28843 — APPROVED** [review]

File: frontend/app/applications/[id]/tabs/InterviewCard.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  rowStack: {
+    flexDirection: 'row',
+    gap: { xs: 1, sm: 2 },
+  },
+  text: {
+    textAlign: { xs: 'center', sm: 'left' },
+  },
+}
+
+const MeetingLink = ({ link }) => (
```
Comment (pankaj28843) [inline frontend/app/applications/[id]/tabs/InterviewCard.js:36]: @olhasashchuk Please make this a more reusable component, `<ExternalLink/>` is a better name and use it in places where you render external links instead of repeating following code:
```
<Link
      href={link}
      target="_blank"
.../>
      ```
